<?xml version="1.0"?>
<launch>
    <arg name="sensor_model" description="Helios|Bpearl|M1"/>
    <arg name="return_mode" default="Strongest" description="See readme for supported return modes"/>
    <arg name="frame_id" default="robosense"/>
    <arg name="scan_phase" default="0.0"/>

    <arg name="calibration_file" default="$(find-pkg-share nebula_decoders)/calibration/robosense/Helios5515.csv"/>

    <arg name="sensor_ip" default="192.168.1.201" description="Lidar Sensor IP"/>
    <arg name="host_ip" default="192.168.1.102" description="Broadcast IP from Sensor"/>
    <arg name="data_port" default="6699" description="LiDAR Data Port"/>
    <arg name="gnss_port" default="7788" description="LiDAR GNSS Port"/>
    <arg name="packet_mtu_size" default="1500" description="Packet MTU size"/>
    <arg name="rotation_speed" default="600" description="Motor RPM, the sensor's internal spin rate."/>
    <arg name="cloud_min_angle" default="0" description="Field of View, start degrees."/>
    <arg name="cloud_max_angle" default="360" description="Field of View, end degrees."/>

    <arg name="setup_sensor" default="True" description="Enable sensor setup on hw-driver."/>
    <arg name="dual_return_distance_threshold" default="0.1" description="Distance threshold of dual return mode"/>
    <arg name="diag_span" default="1000" description="milliseconds"/>


    <node pkg="nebula_ros" exec="robosense_hw_interface_ros_wrapper_node"
          name="robosense_hw_driver" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="data_port" value="$(var data_port)"/>
        <param name="gnss_port" value="$(var gnss_port)"/>
    </node>

    <node pkg="nebula_ros" exec="robosense_driver_ros_wrapper_node"
          name="robosense_cloud" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="gnss_port" value="$(var gnss_port)"/>
        <param name="dual_return_distance_threshold" value="$(var dual_return_distance_threshold)"/>
    </node>

    <node pkg="nebula_ros" exec="robosense_hw_monitor_ros_wrapper_node"
          name="robosense_hw_monitor_$(var frame_id)" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="diag_span" value="$(var diag_span)"/>
    </node>

</launch>